File: domain/src/main/java/com/github/anrimian/musicplayer/domain/repositories/MediaScannerRepository.java
package com.github.anrimian.musicplayer.domain.repositories;

import com.github.anrimian.musicplayer.domain.models.scanner.FileScannerState;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Observable;

public interface MediaScannerRepository {
    void runStorageObserver();

    void rescanStorage();

    Completable runStorageScanner();

    Completable runStorageAndFileScanner();

    Observable<FileScannerState> getFileScannerStateObservable();
}

File: domain/src/main/java/com/github/anrimian/musicplayer/domain/repositories/MediaScannerRepository.kt
package com.github.anrimian.musicplayer.domain.repositories

import com.github.anrimian.musicplayer.domain.models.scanner.FileScannerState
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Observable

interface MediaScannerRepository {

    fun runStorageObserver()

    fun rescanStorage()

    fun rescanStorageAsync()

    fun rescanStoragePlaylists(): Completable

    fun runStorageScanner(): Completable

    fun runStorageAndFileScanner(): Completable

    fun getFileScannerStateObservable(): Observable<FileScannerState>

}